pipeline_dp/histograms.py

```diff
@@ -125,37 +127,66 @@ def _to_bin_lower(n: int) -> int:
     return n // round_base * round_base
 
 
-def _compute_frequency_histogram(col,
-                                 backend: pipeline_backend.PipelineBackend,
-                                 name: HistogramType,
-                                 deduplicate: bool = False):
+def _compute_frequency_histogram(col, backend: pipeline_backend.PipelineBackend,
+                                 name: HistogramType):
     """Computes histogram of element frequencies in collection.
+
     Args:
       col: collection with positive integers.
       backend: PipelineBackend to run operations on the collection.
       name: name which is assigned to the computed histogram.
-      deduplicate: performs deduplication of elements in the histogram. Namely,
-       if true, the histogram (n, count_n) is transformed to (n, count_n/n).
-       It should be used, when collection is generated in a way, that each
-       element n (col contains integers) duplicated n times.
     Returns:
       1 element collection which contains Histogram.
     """
 
     col = backend.count_per_element(col, "Frequency of elements")
-    if deduplicate:
-        col = backend.map_tuple(
-            col, lambda element, frequency:
-            (element, int(round(frequency / element))), "Deduplicate")
 
     # Combiner elements to histogram buckets of increasing sizes. Having buckets
     # of width = 1 is not scalable.
+    return _compute_frequency_histogram_helper(backend, col, name)
+
+
+def _compute_weighted_frequency_histogram(
+        col, backend: pipeline_backend.PipelineBackend, name: HistogramType):
+    """Computes histogram of element frequencies in collection.
+
+    Args:
+      col: collection of (positive integers, weight).
+      backend: PipelineBackend to run operations on the collection.
+      name: name which is assigned to the computed histogram.
+    Returns:
+      1 element collection which contains Histogram.
+    """
+
+    col = backend.sum_per_key(col, "Frequency of elements")
+    # (int, sum_weights:float)
+
+    col = backend.map_values(col, lambda x: int(round(x)), "Round")
+    # (int, sum_weights:int)
+
+    # Combiner elements to histogram buckets of increasing sizes. Having buckets
+    # of width = 1 is not scalable.
+    return _compute_frequency_histogram_helper(backend, col, name)
+
+
+def _compute_frequency_histogram_helper(backend, col, name: HistogramType):
```

Comment:
Please add type hint for backend and have a consistent parameter order here and in the function above